File: 2151-maximum-good-people-based-on-statements/2151-maximum-good-people-based-on-statements.py
class Solution:
    def maximumGood(self, st: List[List[int]]) -> int:
        def go(good):
            for i in range(len(st)):
                if good[i]:
                    for j in range(len(st)):
                        if i==j:
                            continue;
                        if st[i][j]==1 and good[j]==False:
                            return False
                        if st[i][j]==0 and good[j]==True:
                            return False
            return True
        good = [False for x in range(len(st))]
        ans = 0 
        n = 1<<len(st)
        for i in range(n):
            for j in range(len(st)):
                if(i & (1<<j)):
                    good[j]=True
                else:
                    good[j]=False
            if go(good)==True:
                ans = max(ans,good.count(True))
        return ans

File: 2151-maximum-good-people-based-on-statements/2151-maximum-good-people-based-on-statements.cpp
class Solution {
public:
    bool check(bool *good,vector<vector<int>>&st){
        for(int i=0;i<st.size();++i){
            if(good[i]){
                for(int j=0;j<st.size();++j){
                    if(i==j)
                        continue;
                    if(st[i][j]==0 and good[j]==true)
                        return false;
                    if(st[i][j]==1 and good[j]==false)
                        return false;
                }
            }
        }
        return true;
    }
    int maximumGood(vector<vector<int>>& st) {
        int n = 1<<st.size();
        int ans =0;
        bool good[st.size()];
        for (int i=0;i<n;++i){
            for(int j =0;j<st.size();++j){
                if(i & (1<<j)){
                    good[j]=true;
                }else{
                    good[j]=false;
                }
            }
            if(check(good,st)){
                int cnt =0;
                for (int k=0;k<st.size();++k)
                    if(good[k])
                        ++cnt;
                ans = max(ans,cnt);
            }
        }
        return ans;
    }
};